src/ConnectionManager.js

```diff
@@ -28,141 +23,125 @@ export default class ConnectionManager {
       instance: this,
 
       /**
-       * @property {ConnectionQueue} waitingConnections
-       * @description List of waiting connections ordered by priority
-       * @name ConnectionManager~Context#waitingConnections
+       * @property {ConnectionQueue} list
+       * @description List of connections ordered by priority
+       * @name ConnectionManager~Context#list
        */
-      waitingConnections: new ConnectionQueue(),
+      list: List(),
 
       /**
-       * @property {List} openConnections
-       * @description List of open connections
-       * @name ConnectionManager~Context#next
+       * @property {ConnectionQueue} interval
+       * @description Internal loop interval
+       * @name ConnectionManager~Context#interval
        */
-      openConnections: new ConnectionQueue(),
+      interval: null,
 
       /**
-       * @property {function} next
-       * @description Opens the the connection if there's a free slot.
-       * @name ConnectionManager~Context#next
+       * Opens a Connection
+       *
+       * @param {AbstractConnection} connection
        */
-      next() {
-        if (
-          context.openConnections.size >= maxConnections ||
-          context.waitingConnections.size === 0
-        ) {
-          return;
-        }
-
-        const connection = context.waitingConnections.first();
+      setupConnection(connection) {
+        connection.open();
+      },
 
-        if (connection.state === AbstractConnection.INIT) {
-          connection.open(connection.url);
+      /**
+       * This function is where the magic happens.
+       */
+      loop() {
+        // store all open items
+        let openList = context.list.filter(
+          listItem => listItem.connection.state === AbstractConnection.OPEN
+        );
+
+        let waitingList = context.list
+          .filter(
+            listItem => listItem.connection.state === AbstractConnection.INIT
+          )
+          // this code would increase priority on old items, after they lived in the line for more then 30s.
```

Comment:
Is there a good reason this should stay?

Reply:
we have to decide if we want that behaviour or not, thats the reason why i kept it as a comment